Documentation/ja/TestUsingMock.md

```diff
@@ -0,0 +1,153 @@
+# モックを使ったテスト
+
+## モックとは
+
+モックとはテスト対象のオブジェクト(クラス、構造体)が呼び出し先のオブジェクトと意図したとおりに協調動作するかどうかをテストするために使うオブジェクトのことです。
```

Comment:
I prefer the documentation to focus foremost on the benefits of testing applications, and secondly on techniques on how to do so. This explanation of a mock is very technical. It focuses on what a mock is, before explaining what problems a mock would solve. Instead, I would try something like this:

1. Explain a problem that comes up often when people write tests. Here's one example: `BTests` tests `B`, which uses `A`. Now bugs in `A` cause `BTests` to fail, even though `B` is fine! All the failures are confusing--it's hard to answer the question "what's broken?"
2. Explain that we can avoid this problem by using a stand-in for `A` in `BTests`.
3. Show an example of using using a stand-in.
4. Mention that these sorts of "stand-in" objects are referred to as "test doubles", "mocks", and other things. We only introduce these technical details once the user understands the problem.

I think the structure of @jeffh's [Testing in Swift](https://realm.io/news/testing-in-swift/) talk ([Japanese here](https://realm.io/jp/news/testing-in-swift/)) is really good. We should also provide a link to that talk!